core/src/main/java/hudson/Functions.java

```diff
@@ -1512,11 +1512,10 @@ public static String dumpThreadInfo(ThreadInfo ti, ThreadGroupMap map) {
                                              (grp != null ? grp : "?") + " " +
                                              ti.getThreadState());
         if (ti.getLockName() != null) {
-            sb.append(" on " + ti.getLockName());
+            sb.append(" on ").append(ti.getLockName());
```

Comment:
Not sure this is worth the reduced readability?

(Also, can't `javac` do this by now?)

Reply:
I think `javac` will do something automatically if we are >= JDK9. The change might still help with Java 8